src/components/Menu.jsx

```diff
@@ -12,6 +12,8 @@ const propTypes = {
   renderItemLabel: PropTypes.func,
   value: PropTypes.string,
   onSelect: PropTypes.func,
+  shouldTransferFocus: PropTypes.bool,
+  notifyBeforeFocusWrap: PropTypes.func,
 };
```

Comment:
So that code becomes just:
```
if (onBeforeItemFocusChange) {
  focusedIndex = onBeforeItemFocusChange(prevIndex, nextIndex, event.key);
} else {
  focusedIndex = nextIndex;
}
```
That's without any checking of the return value, and I don't even care that much about that. It would be fine to document (um, when there are any API docs at all) that the callback must return null or an integer in the right range, otherwise unspecified bad things will happen.